test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Tests/ActionOverloadingEndToEndTests.cs

```diff
@@ -0,0 +1,266 @@
+    public class ActionOverloadingEndToEndTests : EndToEndTestBase<ActionOverloadingEndToEndTests.TestsStartup>
+    {
+        private readonly Uri _baseUri;
+        private readonly Container _context;
+        public class TestsStartup : TestStartupBase
+        {
+            public override void ConfigureServices(IServiceCollection services)
+            {
+                services.ConfigureControllers(typeof(ProductsController), typeof(OrderLinesController), typeof(PeopleController));
+
+                services.AddControllers().AddOData(opt => opt.Count().Filter().Expand().Select().OrderBy().SetMaxTop(null)
+                    .AddRouteComponents("odata", CommonEndToEndEdmModel.GetEdmModel()));
+            }
+        }
+        public ActionOverloadingEndToEndTests(TestWebApplicationFactory<TestsStartup> fixture)
+            : base(fixture)
+        {
+            _baseUri = new Uri(Client.BaseAddress, "odata/");
+            _context = new Container(_baseUri);
+            _context.HttpClientFactory = HttpClientFactory;
+        }
+
+        [Fact]
+        public void ExecuteOverloadedOperations()
+        {
+            var product = _context.Execute<Common.Clients.EndToEnd.Product>(new Uri(_baseUri + "Products(-10)", UriKind.Absolute)).Single();
+            OperationDescriptor productOperationDescriptor = _context.GetEntityDescriptor(product).OperationDescriptors.Single();
+
+            Assert.Equal("http://localhost/odata/Products(-10)/Default.RetrieveProduct", productOperationDescriptor.Target.AbsoluteUri, true);
+            Assert.Equal("http://localhost/odata/$metadata#Default.RetrieveProduct", productOperationDescriptor.Metadata.AbsoluteUri, true);
+
+            Common.Clients.EndToEnd.OrderLine orderLine = _context.Execute<Common.Clients.EndToEnd.OrderLine>(new Uri(_baseUri + "OrderLines(OrderId=-10,ProductId=-10)", UriKind.Absolute)).Single();
+
+            OperationDescriptor orderLineOperationDescriptor = _context.GetEntityDescriptor(orderLine).OperationDescriptors.Single();
+            Assert.Equal("http://localhost/odata/OrderLines(OrderId=-10,ProductId=-10)/Default.RetrieveProduct", orderLineOperationDescriptor.Target.AbsoluteUri, true);
+            Assert.Equal("http://localhost/odata/$metadata#Default.RetrieveProduct", orderLineOperationDescriptor.Metadata.AbsoluteUri, true);
+
+            // service operation
+            int serviceOperationResult = _context.Execute<int>(new Uri(_baseUri + "RetrieveProduct", UriKind.Absolute), "POST", true).Single();
+            Assert.Equal(-10, serviceOperationResult);
+
+            // actions
+            int productId = _context.Execute<int>(productOperationDescriptor.Target, "POST", true).Single();
+            Assert.Equal(-10, productId);
+
+            int orderLineId = _context.Execute<int>(orderLineOperationDescriptor.Target, "POST", true).Single();
+            Assert.Equal(-10, orderLineId);

```

Comment:
All these different overloads return the same thing. That makes it harder to confirm that different overloads were actually called on the service. Maybe you could have used different products with different IDs for a clearer demonstration.

Reply:
I think the main reason for maintaining some of these values is not to have to change the tests as much and to maintain them as they were before. I guess though we can improve this to make it clearer and simple to understand the intent of the tests.